src/components/panel/panel.js

```diff
@@ -488,33 +523,115 @@ MdPanelRef.prototype.open = function() {
  * closed and animations finish.
  */
 MdPanelRef.prototype.close = function() {
-  if (this.isOpen) {
-    this.isOpen = false;
+  if (!this.isOpen) {
+    // Ignore close. We don't want chaining to happen here.
+    return this._$q.reject('Panel is already closed.');
+  }
 
-    // TODO(ErinCoughlan) - Cancel any in-progress `opening`.
+  if (this._closePromise) {
+    // If the panel is already closed, re-use the (already resolved) promise from
+    // when it was closed.
+    return this._closePromise;
+  }
 
-    if (!this._closePromise) {
-      var self = this;
-      this._closePromise = this._$q(function(resolve, reject) {
-        // TODO(ErinCoughlan): Start the close animation.
-        self._panelContainer.remove();
-        resolve(self);
-      });
+  // TODO(ErinCoughlan) - Cancel any in-progress `opening`.
 
-      this._closePromise.finally(function() {
-        // Clear for next open/close pairing.
-        self._closePromise = undefined;
-      });
+  this.isOpen = false;
+  var self = this;
+  this._closePromise = this._$q(function(resolve, reject) {
+    try {
+      // TODO(ErinCoughlan): Start the close animation.
+      self._panelContainer.remove();
+      resolve(self);
+    } catch(e) {
+      reject(e.message);
     }
-  }
+  });
+
+  this._closePromise.finally(function() {
+    // Clear for next open/close pairing.
+    self._closePromise = undefined;
+  });
 
-  // If the panel is already closed, re-use the (already resolved) promise from
-  // when it was closed.
   return this._closePromise;
 };
 
 
 /**
+ * Show the panel.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * shown and animations finish.
+ */
+MdPanelRef.prototype.show = function() {
```

Comment:
From an API design standpoint, that the difference between open/close and show/hide is rather unclear. I'm thinking perhaps if we make it `setVisible` instead that might be a little more obvious? Thoughts?